packages/drivers/odsp-driver/src/odspFluidFileLink.ts

```diff
@@ -101,13 +101,15 @@ function decodeOdspFluidDataStoreLocator(
 
 	return {
 		siteUrl: siteUrl.href,
-		driveId,
-		itemId,
-		dataStorePath,
-		appName,
-		containerPackageName,
-		fileVersion,
-		context,
+		driveId: decodeURIComponent(driveId),
+		itemId: decodeURIComponent(itemId),
+		dataStorePath: decodeURIComponent(dataStorePath),
+		appName: appName ? decodeURIComponent(appName) : appName,
+		containerPackageName: containerPackageName
+			? decodeURIComponent(containerPackageName)
+			: containerPackageName,
+		fileVersion: fileVersion ? decodeURIComponent(fileVersion) : fileVersion,
+		context: context ? decodeURIComponent(context) : context,
```

Comment:
can you please reference (in comment) the corresponding code that does encoding?

Reply:
I assume this is createOdspUrl(). Doing reverse reference in the code there would be great too.

Reply:
Clearly we are not testing roundtrip with some special characters (like spaces). It would be great to add such coverage as part of this PR

Reply:
We had some tests and added more. 
Actually, UrlSearchParams.get() does the decodeUriComponent by itself, that is why we did not need it here. Removed this change too because of that.